### PR TITLE
Add folding for aten.len.str and E2E test

### PR DESCRIPTION
**Pull request by qedawkins**
New contributor to torch-mlir; starting with adding support for the aten.len.str.

### CODE REVIEW

File: python/torch_mlir_e2e_test/test_suite/basic.py [hunk 1]
```diff
@@ -1959,6 +1959,28 @@ def DetachModule_basic(module, tu: TestUtils):
 # ==============================================================================
 
 
+class StringLength(torch.nn.Module):
```
Comment (vivekkhandelwal1) [inline python/torch_mlir_e2e_test/test_suite/basic.py:1962]: Can you name this something like `LenStrModule`?

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]
Since you have added a folding pattern for this op, could you please add a file check test in https://github.com/llvm/torch-mlir/blob/main/test/Dialect/Torch/canonicalize.mlir as well?

File: test/Dialect/Torch/canonicalize.mlir [hunk 2]
```diff
@@ -457,6 +457,24 @@ func.func @torch.aten.eq.str$same_value() -> !torch.bool {
   return %2 : !torch.bool
 }
 
+// CHECK-LABEL:   func.func @torch.aten.len.str$non_empty() -> !torch.int {
+// CHECK:           %[[INT3:.*]] = torch.constant.int 7
```
Comment (vivekkhandelwal1) [inline test/Dialect/Torch/canonicalize.mlir:461]: nit: `%[[INT3:.*]]` -> `%[[INT7:.*]]`

File: test/Dialect/Torch/canonicalize.mlir [hunk 3]
```diff
@@ -457,6 +457,24 @@ func.func @torch.aten.eq.str$same_value() -> !torch.bool {
...
+// CHECK:           return %[[INT3]] : !torch.int
+func.func @torch.aten.len.str$non_empty() -> !torch.int {
+  %str = torch.constant.str "example"
+  %2 = torch.aten.len.str %str : !torch.str -> !torch.int
+  return %2 : !torch.int
+}
+
+// CHECK-LABEL:   func.func @torch.aten.len.str$empty() -> !torch.int {
+// CHECK:           %[[INT3:.*]] = torch.constant.int 0
```
Comment (vivekkhandelwal1) [inline test/Dialect/Torch/canonicalize.mlir:470]: nit: `%[[INT3:.*]]` -> `%[[INT0:.*]]`

File: test/Dialect/Torch/canonicalize.mlir [hunk 4]
```diff
@@ -457,6 +457,24 @@ func.func @torch.aten.eq.str$same_value() -> !torch.bool {
   return %2 : !torch.bool
 }
 
+// CHECK-LABEL:   func.func @torch.aten.len.str$non_empty() -> !torch.int {
+// CHECK:           %[[INT3:.*]] = torch.constant.int 7
+// CHECK:           return %[[INT3]] : !torch.int
+func.func @torch.aten.len.str$non_empty() -> !torch.int {
```
Comment (vivekkhandelwal1) [inline test/Dialect/Torch/canonicalize.mlir:463]: `@torch.aten.len.str` is sufficient.

**Review by vivekkhandelwal1 — CHANGES_REQUESTED** [review]

**Review by vivekkhandelwal1 — APPROVED** [review]
LGTM

**Review by ramiro050 — APPROVED** [review]
LGTM!

**Pull request merged**